diff --git a/mingw-w64-fribidi/PKGBUILD b/mingw-w64-fribidi/PKGBUILD
index 526b79214..2ab4c484e 100644
--- a/mingw-w64-fribidi/PKGBUILD
+++ b/mingw-w64-fribidi/PKGBUILD
@@ -12,55 +12,42 @@ license=('LGPL')
 url="https://github.com/fribidi/fribidi/"
 depends=()
 options=('strip' '!libtool' 'staticlibs')
-makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
-             "${MINGW_PACKAGE_PREFIX}-meson"
-             "${MINGW_PACKAGE_PREFIX}-ninja")
-source=(${_realname}-${pkgver}.tar.gz::"https://github.com/fribidi/fribidi/archive/v${pkgver}.tar.gz")
-sha256sums=('3ebb19c4184ed6dc324d2e291d7465bc6108a20be019f053f33228e07e879c4f')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc")
+source=(${_realname}-${pkgver}.tar.gz::"https://github.com/fribidi/fribidi/archive/v${pkgver}.tar.gz"
+        "0001-disable-build-doc.patch"
+        "0002-Fix-issue-156-in-fribidi.patch")
+sha256sums=('3ebb19c4184ed6dc324d2e291d7465bc6108a20be019f053f33228e07e879c4f'
+            'c50855ea302951dc0564aca95f0e7cf59d1f8ebfaf8ff833a723548615341a02'
+            '388d94129d316ae2f7e712b0235a88aeac6317596e3a2c94c9579070d8c0e143')
+
+prepare() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  patch -p1 -i ${srcdir}/0001-disable-build-doc.patch
+  patch -p1 -i ${srcdir}/0002-Fix-issue-156-in-fribidi.patch
+  WANT_AUTOMAKE=latest autoreconf -fiv -I m4
+}
 
 build() {
-  mkdir -p build-${MINGW_CHOST}-static
-  cd build-${MINGW_CHOST}-static
+  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
+  mkdir -p "${srcdir}/build-${MINGW_CHOST}" && cd "${srcdir}/build-${MINGW_CHOST}"
 
-  MSYS2_ARG_CONV_EXCL="--prefix=" \
-  meson \
-      --prefix=${MINGW_PREFIX} \
-      --buildtype plain \
+  ../${_realname}-${pkgver}/configure \
       --prefix=${MINGW_PREFIX} \
-       -Ddocs=false \
-       --default-library static \
-      ../${_realname}-${pkgver}
-
-  ninja
+    --build=${MINGW_CHOST} \
+    --host=${MINGW_CHOST} \
+    --target=${MINGW_CHOST} \
+    --enable-shared=no \
+    --enable-static=yes
 
-  cd ..
-  mkdir -p build-${MINGW_CHOST}-shared
-  cd build-${MINGW_CHOST}-shared
-
-  MSYS2_ARG_CONV_EXCL="--prefix=" \
-  meson \
-      --prefix=${MINGW_PREFIX} \
-      --buildtype plain \
-      --prefix=${MINGW_PREFIX} \
-       -Ddocs=false \
-       --default-library shared \
-      ../${_realname}-${pkgver}
-
-  ninja
-}
-
-check() {
-  cd build-${MINGW_CHOST}-shared
-
-  meson test
+  make
 }
 
 package() {
-  cd ${srcdir}/build-${MINGW_CHOST}-static
-  DESTDIR="${pkgdir}" ninja install
-
-  cd ${srcdir}/build-${MINGW_CHOST}-shared
-  DESTDIR="${pkgdir}" ninja install
+  cd ${srcdir}/build-${MINGW_CHOST}
+  make DESTDIR="${pkgdir}" install
 
+  find "${pkgdir}${MINGW_PREFIX}" -name '*.def' -o -name '*.exp' | xargs -rtl1 rm
+  local PREFIX_DEPS=$(cygpath -am ${MINGW_PREFIX})
+  find ${pkgdir}${MINGW_PREFIX}/lib/pkgconfig -name *.pc -exec sed -i -e"s|${PREFIX_DEPS}|${MINGW_PREFIX}|g" {} \;
   install -Dm644 "${srcdir}/${_realname}-${pkgver}/COPYING" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/COPYING"
 }
